app/app.py

```diff
@@ -110,9 +113,57 @@ def callback():
     data = json.loads(res)
 
     session['display_name'] = data['display_name']
+    session['spotify_client_id'] = SPOTIFY_CLIENT_ID
 
     return redirect(url_for('home'))
 
+@app.route('/higher_lower')
+def higher_lower():
+    dummysongs = os.path.dirname(os.path.abspath(__file__)) + '/dummysongs.json'
+    fin = open(dummysongs, 'r')
+    dummySongsJSON = fin.readlines()
+    dummySongsJSON = json.loads(dummySongsJSON[0])
+    dummySongs = dummySongsJSON['items']
+    songs = list()
+    for song in dummySongs:
+        songData = {
+            'title': song['track']['name'],
+            'artist': song['track']['album']['artists'][0]['name'],
+            'coverArtLink': song['track']['album']['images'][0]['url'],
+            'popularity': song['track']['popularity'],
+            'iframe': f"{song['track']['external_urls']['spotify'][:25]}embed/{song['track']['external_urls']['spotify'][25:]}",
+            'spotify_id': song['track']['id']
+        }
+        songs.append(songData)
+
+    return render_template(
+        'higherlowergame.html',
+        songs=songs
+        )
+
+@app.route("/save_song/<song_id>")
+def save_song(song_id):
+    if not 'access_token' in session:
+        flash('You are not connected to your Spotify account', 'error')
+        return redirect(url_for('home'))
```

Comment:
@login_required decorator

Reply:
the access token check acts as a login decorator

Reply:
okie